src/v/cluster/topics_frontend.cc

```diff
@@ -270,6 +293,19 @@ ss::future<std::vector<topic_result>> topics_frontend::update_topic_properties(
 
         auto results = co_await ssx::parallel_transform(
           std::move(updates), [this, timeout](topic_properties_update update) {
+              if (
+                _features.local().should_sanction()
+                && is_user_topic(update.tp_ns)) {
+                  if (auto f = get_enterprise_features(_metadata_cache, update);
```

Comment:
now that "error handling" for sanctioning enterprise features is in the `topics_frontend`, should [schema validation](https://github.com/redpanda-data/redpanda/blob/28f489744352433f2c4c62241f916350d2803e6d/src/v/cluster/topic_table.cc#L1066) and [multi-property validation](https://github.com/redpanda-data/redpanda/blob/28f489744352433f2c4c62241f916350d2803e6d/src/v/cluster/topic_table.cc#L1070) also find its way into the `frontend` instead of the `topic_table`?

I don't have a strong argument for either, but it would be nice to see the various forms of validation kept closer together than they are today.

Reply:
> I don't have a strong argument for either, but it would be nice to see the various forms of validation kept closer together than they are today.

Topic properties could do with a good refactor, yes. It may have to wait for a followup, though.